python/ray/workflow/tests/test_basic_workflows_2.py

```diff
@@ -22,9 +24,11 @@ def test_init_twice(call_ray_start, reset_workflow, tmp_path):
 
 
 def test_init_twice_2(call_ray_start, reset_workflow, tmp_path):
-    run_string_as_driver(driver_script)
-    with pytest.raises(RuntimeError):
-        workflow.init(str(tmp_path))
+    with patch.dict(os.environ, {"RAY_ADDRESS": call_ray_start}):
+        run_string_as_driver(driver_script)
+        with pytest.raises(
+                RuntimeError, match=".*different from the workflow manager.*"):
+            workflow.init(str(tmp_path))
```

Comment:
FYI, this test previously relied on Ray's `auto_init` functionality using a file to realize there is a running cluster. This switches to using `RAY_ADDRESS`, the more widely used method for ensuring `ray.init()` connects to the existing cluster.

Reply:
This look ok to me, cc @iycheng